In [1]:
import pandas as pd 
import numpy as np 
import seaborn as sns

In [3]:
forestfire=pd.read_csv('forestfires (3).csv')
forestfire.head()

month  day  FFMC   DMC     DC  ISI  temp  RH  wind  rain  ...  monthfeb  \
0   mar  fri  86.2  26.2   94.3  5.1   8.2  51   6.7   0.0  ...         0   
1   oct  tue  90.6  35.4  669.1  6.7  18.0  33   0.9   0.0  ...         0   
2   oct  sat  90.6  43.7  686.9  6.7  14.6  33   1.3   0.0  ...         0   
3   mar  fri  91.7  33.3   77.5  9.0   8.3  97   4.0   0.2  ...         0   
4   mar  sun  89.3  51.3  102.2  9.6  11.4  99   1.8   0.0  ...         0   

   monthjan  monthjul  monthjun  monthmar  monthmay  monthnov  monthoct  \
0         0         0         0         1         0         0         0   
1         0         0         0         0         0         0         1   
2         0         0         0         0         0         0         1   
3         0         0         0         1         0         0         0   
4         0         0         0         1         0         0         0   

   monthsep  size_category  
0         0          small  
1         0          small  
2         0          small  
3         0          small  
4         0          small  

[5 rows x 31 columns]

In [4]:
forestfire.describe()

FFMC         DMC          DC         ISI        temp          RH  \
count  517.000000  517.000000  517.000000  517.000000  517.000000  517.000000   
mean    90.644681  110.872340  547.940039    9.021663   18.889168   44.288201   
std      5.520111   64.046482  248.066192    4.559477    5.806625   16.317469   
min     18.700000    1.100000    7.900000    0.000000    2.200000   15.000000   
25%     90.200000   68.600000  437.700000    6.500000   15.500000   33.000000   
50%     91.600000  108.300000  664.200000    8.400000   19.300000   42.000000   
75%     92.900000  142.400000  713.900000   10.800000   22.800000   53.000000   
max     96.200000  291.300000  860.600000   56.100000   33.300000  100.000000   

             wind        rain         area      dayfri  ...    monthdec  \
count  517.000000  517.000000   517.000000  517.000000  ...  517.000000   
mean     4.017602    0.021663    12.847292    0.164410  ...    0.017408   
std      1.791653    0.295959    63.655818    0.371006  ...    0.130913   
min      0.400000    0.000000     0.000000    0.000000  ...    0.000000   
25%      2.700000    0.000000     0.000000    0.000000  ...    0.000000   
50%      4.000000    0.000000     0.520000    0.000000  ...    0.000000   
75%      4.900000    0.000000     6.570000    0.000000  ...    0.000000   
max      9.400000    6.400000  1090.840000    1.000000  ...    1.000000   

         monthfeb    monthjan    monthjul    monthjun    monthmar    monthmay  \
count  517.000000  517.000000  517.000000  517.000000  517.000000  517.000000   
mean     0.038685    0.003868    0.061896    0.032882    0.104449    0.003868   
std      0.193029    0.062137    0.241199    0.178500    0.306138    0.062137   
min      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
50%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
75%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
max      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   

         monthnov    monthoct    monthsep  
count  517.000000  517.000000  517.000000  
mean     0.001934    0.029014    0.332689  
std      0.043980    0.168007    0.471632  
min      0.000000    0.000000    0.000000  
25%      0.000000    0.000000    0.000000  
50%      0.000000    0.000000    0.000000  
75%      0.000000    0.000000    1.000000  
max      1.000000    1.000000    1.000000  

[8 rows x 28 columns]

In [5]:
forestfire.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 517 entries, 0 to 516
Data columns (total 31 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   month          517 non-null    object 
 1   day            517 non-null    object 
 2   FFMC           517 non-null    float64
 3   DMC            517 non-null    float64
 4   DC             517 non-null    float64
 5   ISI            517 non-null    float64
 6   temp           517 non-null    float64
 7   RH             517 non-null    int64  
 8   wind           517 non-null    float64
 9   rain           517 non-null    float64
 10  area           517 non-null    float64
 11  dayfri         517 non-null    int64  
 12  daymon         517 non-null    int64  
 13  daysat         517 non-null    int64  
 14  daysun         517 non-null    int64  
 15  daythu         517 non-null    int64  
 16  daytue         517 non-null    int64  
 17  daywed         517 non-null    int64  
 18  monthapr  

In [7]:
## Dropping month & day from dataframe
forestfire.drop(["month","day"],axis=1,inplace =True)

In [8]:
forestfire.head()

FFMC   DMC     DC  ISI  temp  RH  wind  rain  area  dayfri  ...  monthfeb  \
0  86.2  26.2   94.3  5.1   8.2  51   6.7   0.0   0.0       1  ...         0   
1  90.6  35.4  669.1  6.7  18.0  33   0.9   0.0   0.0       0  ...         0   
2  90.6  43.7  686.9  6.7  14.6  33   1.3   0.0   0.0       0  ...         0   
3  91.7  33.3   77.5  9.0   8.3  97   4.0   0.2   0.0       1  ...         0   
4  89.3  51.3  102.2  9.6  11.4  99   1.8   0.0   0.0       0  ...         0   

   monthjan  monthjul  monthjun  monthmar  monthmay  monthnov  monthoct  \
0         0         0         0         1         0         0         0   
1         0         0         0         0         0         0         1   
2         0         0         0         0         0         0         1   
3         0         0         0         1         0         0         0   
4         0         0         0         1         0         0         0   

   monthsep  size_category  
0         0          small  
1         0          small  
2         0          small  
3         0          small  
4         0          small  

[5 rows x 29 columns]

In [11]:
## Locating predictors & targets
predictors = forestfire.iloc[:,0:28]
target = forestfire.iloc[:,28]

In [12]:
## normalizing function
def norm_func(i):
    x= (i-i.min())/(i.max()-i.min())
    return (x)

In [18]:
forestfire_n = norm_func(predictors)

In [19]:
forestfire_n.head()

FFMC       DMC        DC       ISI      temp        RH      wind  \
0  0.870968  0.086492  0.101325  0.090909  0.192926  0.423529  0.700000   
1  0.927742  0.118194  0.775419  0.119430  0.508039  0.211765  0.055556   
2  0.927742  0.146795  0.796294  0.119430  0.398714  0.211765  0.100000   
3  0.941935  0.110958  0.081623  0.160428  0.196141  0.964706  0.400000   
4  0.910968  0.172984  0.110590  0.171123  0.295820  0.988235  0.155556   

      rain  area  dayfri  ...  monthdec  monthfeb  monthjan  monthjul  \
0  0.00000   0.0     1.0  ...       0.0       0.0       0.0       0.0   
1  0.00000   0.0     0.0  ...       0.0       0.0       0.0       0.0   
2  0.00000   0.0     0.0  ...       0.0       0.0       0.0       0.0   
3  0.03125   0.0     1.0  ...       0.0       0.0       0.0       0.0   
4  0.00000   0.0     0.0  ...       0.0       0.0       0.0       0.0   

   monthjun  monthmar  monthmay  monthnov  monthoct  monthsep  
0       0.0       1.0       0.0       0.0       0.0       0.0  
1       0.0       0.0       0.0       0.0       1.0       0.0  
2       0.0       0.0       0.0       0.0       1.0       0.0  
3       0.0       1.0       0.0       0.0       0.0       0.0  
4       0.0       1.0       0.0       0.0       0.0       0.0  

[5 rows x 28 columns]

In [20]:
## importing Support Vector Machine

from sklearn.svm import SVC
from sklearn.model_selection import train_test_split

In [21]:
x_train,x_test,y_train,y_test = train_test_split(predictors,target,test_size = 0.20, stratify = target)

In [22]:
x_train

FFMC    DMC     DC   ISI  temp  RH  wind  rain   area  dayfri  ...  \
243  91.8  175.1  700.7  13.8  21.9  73   7.6   1.0   0.00       0  ...   
270  92.1  152.6  658.2  14.3  21.8  56   3.1   0.0   0.52       0  ...   
202  83.9    8.7   32.1   2.1   8.8  68   2.2   0.0  13.05       0  ...   
287  91.6  104.2  474.9   9.0  22.1  49   2.7   0.0   0.00       0  ...   
76   86.6   13.2   43.0   5.3  15.7  43   3.1   0.0   0.00       1  ...   
..    ...    ...    ...   ...   ...  ..   ...   ...    ...     ...  ...   
80   91.4  142.4  601.4  10.6  19.5  39   6.3   0.0   0.00       0  ...   
20   91.0  129.5  692.6   7.0  18.3  40   2.7   0.0   0.00       0  ...   
59   86.6   13.2   43.0   5.3  12.3  51   0.9   0.0   0.00       1  ...   
483  94.9  130.3  587.1  14.1  31.0  27   5.4   0.0   0.00       0  ...   
298  91.2  147.8  377.2  12.7  19.6  43   4.9   0.0   0.00       0  ...   

     monthdec  monthfeb  monthjan  monthjul  monthjun  monthmar  monthmay  \
243         0         0         0         0         0         0         0   
270         0         0         0         0         0         0         0   
202         0         1         0         0         0         0         0   
287         0         0         0         1         0         0         0   
76          0         1         0         0         0         0         0   
..        ...       ...       ...       ...       ...       ...       ...   
80          0         0         0         0         0         0         0   
20          0         0         0         0         0         0         0   
59          0         1         0         0         0         0         0   
483         0         0         0         0         0         0         0   
298         0         0         0         0         1         0         0   

     monthnov  monthoct  monthsep  
243         0         0         0  
270         0         0         0  
202         0         0         0  
287         0         0         0  
76          0         0         0  
..        ...       ...       ...  
80          0         0         0  
20          0         0         1  
59          0         0         0  
483         0         0         0  
298         0         0         0  

[413 rows x 28 columns]

In [23]:
x_test

FFMC    DMC     DC   ISI  temp  RH  wind  rain   area  dayfri  ...  \
373  94.8  222.4  698.6  13.9  20.3  42   2.7   0.0   0.00       0  ...   
369  91.0  276.3  825.1   7.1  13.8  77   7.6   0.0   0.00       0  ...   
186  89.3   51.3  102.2   9.6  11.5  39   5.8   0.0   7.19       0  ...   
459  91.6  273.8  819.1   7.7  15.5  72   8.0   0.0   1.94       0  ...   
381  90.7  194.1  643.0   6.8  16.2  63   2.7   0.0  16.33       0  ...   
..    ...    ...    ...   ...   ...  ..   ...   ...    ...     ...  ...   
474  93.0  103.8  316.7  10.8  26.4  35   2.7   0.0  10.08       0  ...   
386  90.5   96.7  750.5  11.4  20.6  55   5.4   0.0  24.59       0  ...   
470  91.0   14.6   25.6  12.3  17.6  27   5.8   0.0   0.00       0  ...   
44   90.1   82.9  735.7   6.2  12.9  74   4.9   0.0   0.00       0  ...   
286  91.2  183.1  437.7  12.5  12.6  90   7.6   0.2   0.00       0  ...   

     monthdec  monthfeb  monthjan  monthjul  monthjun  monthmar  monthmay  \
373         0         0         0         0         0         0         0   
369         0         0         0         0         0         0         0   
186         0         0         0         0         0         1         0   
459         0         0         0         0         0         0         0   
381         0         0         0         0         0         0         0   
..        ...       ...       ...       ...       ...       ...       ...   
474         0         0         0         0         1         0         0   
386         0         0         0         0         0         0         0   
470         0         0         0         0         0         0         0   
44          0         0         0         0         0         0         0   
286         0         0         0         1         0         0         0   

     monthnov  monthoct  monthsep  
373         0         0         0  
369         0         0         1  
186         0         0         0  
459         0         0         0  
381         0         0         0  
..        ...       ...       ...  
474         0         0         0  
386         0         0         1  
470         0         0         0  
44          0         0         1  
286         0         0         0  

[104 rows x 28 columns]

In [35]:
## using linear kernel SVM

model_linear = SVC(kernel = "linear")
model_linear.fit(x_train,y_train)
pred_test_linear = model_linear.predict(x_test)

In [36]:
np.mean(pred_test_linear==y_test) 

0.9903846153846154

Accuracy for training data using linear Kernel SVM is 99% accurate

In [38]:
## using linear kernel SVM

model_poly = SVC(kernel = "poly")
model_poly.fit(x_train,y_train)
pred_test_poly = model_poly.predict(x_test)

In [39]:
np.mean(pred_test_poly==y_test)

0.75

Accuracy for test data using poly Kernel SVM is 75% accurate

In [41]:
## using rbf kernel SVM

model_rbf = SVC(kernel = "rbf")
model_rbf.fit(x_train,y_train)
pred_test_rbf = model_rbf.predict(x_test)

In [42]:
np.mean(pred_test_rbf==y_test)

0.7403846153846154

Accuracy for test data using rbf Kernel SVM is 74% accurate

# For this data, using linear kernel SVM has a 99% accuracy in predicting forest fires.